### PR TITLE
Add link checker and fix broken link logic

### PR DESCRIPTION
**Pull request by lucperkins**
This PR adds a link checker to the site build that identifies broken internal links.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,9 +11,22 @@ production-build:
...
 	hugo \
 		--baseURL $(DEPLOY_PRIME_URL) \
 		--buildDrafts \
 		--buildFuture \
 		--minify
+
+	make check-internal-links
+
+install-link-checker:
+	curl https://raw.githubusercontent.com/wjdp/htmltest/master/godownloader.sh | bash
```
Comment (duglin) [inline Makefile]: I'm not thrilled with the idea of the Makefile downloading stuff into someone's machine.

Comment (duglin) [reply Makefile]: While it's not the fastest, at this the one we use for the spec is safe: https://github.com/cloudevents/spec/blob/master/Makefile#L6

**Comment by lucperkins** [comment]
@duglin This is designed above all to be applied at the CI level. I'd be happy to add appropriate warnings in the README and/or add the script to source control to eliminate the curl-to-bash.

The solution used for the spec is a minimalistic one that doesn't provide checks for static assets, alt tags in images, etc. I think we'd be better off with the status quo than adopting that solution. Let me know what you think.

**Comment by duglin** [comment]
it may be intended to be a CI thing, but with it being in the Makefile I fear someone will run it w/o thinking about it.

@markpeek what are your thoughts?

**Comment by markpeek** [comment]
I'm in agreement with @duglin, we should not download/install/run arbitrary binaries via the Makefile. The Makefile (or a script) should check for the existence of the binary prior to running it with an appropriate message about how to download it. When CI is implemented, the binary can be downloaded automatically in the context of that CI step. This protects endusers from unintentional consequences from running make.

**Comment by lucperkins** [comment]
We already have a CI system in place

**Pull request closed without merging**